src/main/java/org/broadinstitute/hellbender/tools/walkers/longreads/RepairLongReadBam.java

```diff
@@ -0,0 +1,131 @@
+import org.broadinstitute.barclay.argparser.BetaFeature;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.argparser.ExperimentalFeature;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import picard.cmdline.programgroups.ReadDataManipulationProgramGroup;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.ReadWalker;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * Restore annotations from unaligned BAM files that are discarded during conversion by samtools fastq
+ *
+ * <h3>Input</h3>
+ * <ul>
+ *     <li>A unaligned BAM file</li>
+ *     <li>A aligned BAM file</li>
+ * </ul>
+ *
+ * <h3>Outputs</h3>
+ * <ul>
+ *     <li>An aligned BAM file with annotations restored</li>
+ * </ul>
+ *
+ * <h3>Usage Example</h3>
+ * <h4>Restore annotations from unaligned BAM files that may be discarded during conversion by samtools fastq</h4>
+ * <pre>
+ *   gatk RepairLongReadBam \
+ *     -I unaligned.bam \
+ *     -A aligned.bam \
+ *     -O restored.bam
+ * </pre>
+ */
+@DocumentedFeature
+@ExperimentalFeature
+@CommandLineProgramProperties(
+        summary = "Restore annotations from unaligned BAM files that are discarded during conversion by samtools fastq",
+        oneLineSummary = "Restore annotations from unaligned BAM files that are discarded during conversion by samtools fastq",
+        programGroup = ReadDataManipulationProgramGroup.class
+)
+public final class RepairLongReadBam extends ReadWalker {
+    @Argument(fullName = "aligned", shortName = "A", doc="aligned reads")
+    public String aligned;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc="Write output to this file")
+    public String output;
+
+    @Argument(fullName = "sort", shortName = "s", doc="Sort output", optional = true)
+    public boolean sort = false;
+
+    private SAMFileWriter writer;
+    private SAMRecordIterator it;
+    private SAMRecord currentAlignedRead;
+
+    @Override
+    public void onTraversalStart() {
+        if ( readArguments.getReadPathSpecifiers().size() != 1 ) {
+            throw new UserException("Specify a single unaligned BAM file to -I and a single aligned BAM file to -A");
+        }
+
+        SamReaderFactory srf = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.SILENT);
+        SamReader srs = srf.open(IOUtils.getPath(aligned));
+
+        if (getHeaderForReads() != null && getHeaderForReads().getReadGroups() != null && getHeaderForReads().getReadGroups().size() != 1) {
```

Comment:
also add a check that both of these read group headers have the SAME read group, not just that they are both unique.

Reply:
a check worth considering is to check if hte sort orders in the headers match.